common/djangoapps/student/management/tests/test_bulk_change_enrollment.py

```diff
@@ -0,0 +1,103 @@
+"""Tests for the bulk_change_enrollment command."""
+import ddt
+from django.core.management import call_command
+from django.core.management.base import CommandError
+
+from student.tests.factories import UserFactory, CourseModeFactory, CourseEnrollmentFactory
+from student.models import CourseEnrollment
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+
+
+@ddt.ddt
+class BulkChangeEnrollmentTests(SharedModuleStoreTestCase):
+    """Tests for the bulk_change_enrollment command."""
+
+    def setUp(self):
+        super(BulkChangeEnrollmentTests, self).setUp()
+        self.course = CourseFactory.create()
+        self.users = UserFactory.create_batch(5)
+
+    @ddt.data(('audit', 'honor'), ('honor', 'audit'))
+    @ddt.unpack
+    def test_bulk_convert(self, from_mode, to_mode):
+        """Verify that enrollments are changed correctly."""
+        self._enroll_users(from_mode)
+        CourseModeFactory(course_id=self.course.id, mode_slug=to_mode)
+
+        # Verify that no users are in the `from` mode yet.
+        self.assertEqual(len(CourseEnrollment.objects.filter(mode=to_mode, course_id=self.course.id)), 0)
+
+        call_command(
+            'bulk_change_enrollment',
+            course=unicode(self.course.id),
+            from_mode=from_mode,
+            to_mode=to_mode,
+            commit=True,
+        )
+
+        # Verify that all users have been moved
```

Comment:
Or rather, I would prefer a true assertion here.  I can see that we are probably relying on DoesNotExist to blow up this test.